app/controllers/memberships_controller.rb

```diff
@@ -0,0 +1,42 @@
+class MembershipsController < ApplicationController
+
+  def join
+    @organization = Organization.find membership_params[:organization_id]
+    @membership = Membership.new
+    @membership.user = current_user
+    @membership.organization = @organization
+
+    if @membership.save
+      flash[:success] = "Membership request sent."
+    else
+      flash[:danger] = @membership.errors.full_messages
+    end
+
+    redirect_to @organization
```

Comment:
Controller actions should only ever instantiate one instance variable, which is the instance variable that will be used in the view. Since we're not rendering the membership here, that should be `@organization`.

I would also advocate for more concision here, and would write the first part as:

```ruby
@organization = Organization.find membership_params[:organization_id]
membership = Membership.new(organization: @organization, user: current_user)
```

There are a few other spots in this file that can be condensed.